scripts/enforce-eol.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# define command as the script parameter or default to check
+command=${1:-check}
+
+directories=("lib" "scripts" "docs" "nix" "specifications" "configs")
+
+regex=".*\.\(md\|hs\|sh\|nix\|yaml\|yml\|toml\|cabal\)$"
+
+check_eol() {
+    local directory=$1
+    find "$directory" -type f -regex "$regex" |
+        while read -r file; do
+            if [ -s "$file" ]; then
+                if [ "$(tail -c 1 "$file")" != "" ]; then
+                    echo "File $file does not end with a newline character."
+                    echo "Fix this by running ./scripts/enforce-eol.sh add"
+                    exit 1
+                fi
+            fi
+        done
+}
+
+add_eol() {
+    local directory=$1
+    # shellcheck disable=SC286
```

Comment:
Is this check needed? I could not find a reference to SC286, only SC2086 which is about quoting